pippy/PipelineSchedule.py

```diff
@@ -131,11 +121,51 @@ def __init__(
             f"""
             finished pipeline stage init, {self.stage_id=}, {self.is_first_stage=},
             {self.is_last_stage=}, {self.num_stages=},
-            {[fwd_input.shape for fwd_input in self.fwd_inputs]},
-            {[fwd_output_grad.shape for fwd_output_grad in self.fwd_output_grads]}
             """
         )
 
+    def to(self, *args, **kwargs):
+        """
+        Move the module to a new device or data type, including the buffers.
+        """
+        super().to(*args, **kwargs)
+
+        # find the device of the underlying module and move the buffers to it if they are meta
+        device = next(self.module.parameters()).device
+
+        for i, fwd_input in enumerate(self.fwd_inputs):
+            if fwd_input.is_meta:
+                self.fwd_inputs[i] = torch.empty_like(fwd_input, device=device)
+            self.fwd_inputs[i] = self.fwd_inputs[i].to(*args, **kwargs)
+        for i, fwd_output_grad in enumerate(self.fwd_output_grads):
+            if fwd_output_grad.is_meta:
+                self.fwd_output_grads[i] = torch.empty_like(
+                    fwd_output_grad, device=device
+                )
+            self.fwd_output_grads[i] = self.fwd_output_grads[i].to(
+                *args, **kwargs
+            )
+
+    def create_buffers(
+        self, inputs_meta: List[torch.tensor]
+    ) -> List[torch.Tensor]:
+        """
+        Creates buffers for a given input on a specified device.
+        This function takes as input a meta tensor or a list of meta tensors
+        and returns a flattened list of empty tensors of the same shape.
+        """
+        if isinstance(inputs_meta, torch.Tensor):
```

Comment:
It seems this `if` branch would never be true based on the signature of this function.